src/client/mod.rs

```diff
@@ -52,6 +52,37 @@ use std::{
     fmt,
 };
 
+/// Our LazyMesssage error. Recipient was unable to process this message for some reason.
+/// The original message should be returned in full, and context can optionally be added via
+/// reason.
+pub struct CannotProcessMessage {
+    reason: Option<CannotProcessMessageReason>,
+    source_message: Message,
+}
+
+/// Error reasons for inability to handle a message at a client/node.
+/// These should be recoverable with updated information from the sender
+/// or other nodes in the network.
+///
+/// A lot of errors should be handled even before they reach the node via
+/// Routing ensuring we're targetting the correct section key.
+pub enum CannotProcessMessageReason {
```

Comment:
It seems all the `reasons` are raised by `nodes` and be received by `node` or `client`.
Then shall it be defined in a common file instead of within `client/mod.rs` ?

Reply:
They could be raised by client's as well.

There is an intention to split up the messaging for node/client somewhat. But I'd opt to leave that for now